### PR TITLE
[Maintenance] Rector version upgrade, static analysis included

### PR DESCRIPTION
**Pull request by Rafikooo**
As Sylius-Standard currently depends on `^0.14` rector version it is impossible to install `SyliusRector` without playing around with dependency versions.

image

### CODE REVIEW

**Review by coldic3 — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "description": "Rector upgrades rules for Sylius",
     "require": {
         "php": "^8.0",
-        "rector/rector": "^0.13"
```
Comment (jakubtobiasz) [inline composer.json:8]: Is it problematic to support ^0.13 and ^0.14 at the same time?

Comment (Rafikooo) [reply composer.json:8]: I suppose the problem is only with testing as the API of RectorTestCase has changed between [^0.14](https://github.com/rectorphp/rector/blob/0.14.8/packages/Testing/PHPUnit/AbstractRectorTestCase.php#L81) and [^0.13](https://github.com/rectorphp/rector/blob/0.13.10/packages/Testing/PHPUnit/AbstractRectorTestCase.php#L91)

File: tests/Rector/Class_/AddInterfaceToClassExtendingType/AddInterfaceToClassExtendingTypeTest.php [hunk 2]
```diff
@@ -6,20 +6,19 @@
...
-    public function test(SmartFileInfo $fileInfo): void
+    public function test(string $file): void
     {
-        $this->doTestFileInfo($fileInfo);
+        $this->doTestFile($file);
     }
 
     /**
-     * @return Iterator<SmartFileInfo>
+     * @return Iterator<string>
```
Comment (Rafikooo) [inline tests/Rector/Class_/AddInterfaceToClassExtendingType/AddInterfaceToClassExtendingTypeTest.php:21]: Wondering with
image
vs

[this](https://github.com/rectorphp/rector/blob/0.14.8/packages/Testing/PHPUnit/AbstractRectorTestCase.php#L81)

**Review by jakubtobiasz — APPROVED** [review]

**Review by Zales0123 — APPROVED** [review]

**Comment by Zales0123** [comment]
Thank you, @Rafikooo! 🎉

**Pull request merged**